### PR TITLE
Closes #249 #64 - Added observer configuration validation and improved e...

### LINKED ISSUE #249: Initialization silent on bad input
@joshelser tried to initialize phrasecount using bad config (before fixing fluo-io/phrasecount#8) and init did not fail.  See [email thread][1] for prop file with bad input.

[1]: https://groups.google.com/d/msgid/fluo-dev/a3263273-873b-4c06-b06c-87c23096e004%40googlegroups.com?utm_medium=email&utm_source=footer

### PR DESCRIPTION
**Pull request by mikewalch**
...rror reporting

### CODE REVIEW

**Comment by mikewalch** [comment]
In a separate PR for #293, I will add argument checking for other properties in `FluoConfiguration`

File: modules/api/src/test/java/io/fluo/api/config/FluoConfigurationTest.java [hunk 1]
```diff
@@ -201,4 +201,44 @@ public void testMultipleZookeepers(){
...
+  @Test
+  public void testObserverConfig() {
+    FluoConfiguration config = new FluoConfiguration();
+    
+    config.setProperty("key1", "com.foo.Observer2,configKey1=configVal1,configKey2=configVal2");
+    
+    ObserverConfiguration oc = config.getObserverConfig("key1");
+    Assert.assertEquals("com.foo.Observer2", oc.getClassName());
+    Assert.assertEquals("configVal1", oc.getParameters().get("configKey1"));
+    Assert.assertEquals(2, oc.getParameters().size());
```
Comment (keith-turner) [inline modules/api/src/test/java/io/fluo/api/config/FluoConfigurationTest.java]: whay not validate configKey2?

File: modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java [hunk 2]
```diff
@@ -269,6 +280,39 @@ public FluoConfiguration setWorkerThreads(int numThreads) {
   public int getWorkerThreads() {
     return getInt(WORKER_NUM_THREADS_PROP, WORKER_NUM_THREADS_DEFAULT);
   }
+  
+  public ObserverConfiguration getObserverConfig(String key) {
```
Comment (keith-turner) [inline modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java]: could make this mirror the set method and have `List<ObserverConfiguration> getObservers()` instead

Comment (mikewalch) [inline modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java]: Good idea.  I made the get method only return one observer as it fit better with the existing implementation of `FluoAdminImpl`.  However, I think refactoring `FluoAdminImpl` to use a get method that returns a `List` will help readability.  I'll change this.

Comment (mikewalch) [inline modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java]: updated in ab8eb477ddcb19

**Comment by keith-turner** [comment]
+1

**Pull request merged**